pulp_container/app/registry_api.py

```diff
@@ -785,19 +790,33 @@ def create_blob(self, artifact, digest):
             except IntegrityError:
                 blob = models.Blob.objects.get(digest=digest)
                 blob.touch()
-            try:
-                blob_artifact = ContentArtifact(
-                    artifact=artifact, content=blob, relative_path=digest
-                )
-                blob_artifact.save()
-            except IntegrityError:
-                # re-upload artifact in case it was previously removed.
-                ca = ContentArtifact.objects.get(content=blob, relative_path=digest)
-                if not ca.artifact:
-                    ca.artifact = artifact
-                    ca.save(update_fields=["artifact"])
+
+            blob_size = artifact.file.size
```

Comment:
This is comples part of code, however, I was wondering if it worth investing time so we do not create unnecessary Artifact if the size is small enough. We first initialize PulpTemporaryUploadedFile which has sizr info we could use and then depending on that create or not the Artifact.
(might be too complicated but anyway have thrown this idea on the table)